lib/Transport.js

```diff
@@ -51,6 +51,7 @@ const clientVersion = require('../package.json').version
 const userAgent = `opensearch-js/${clientVersion} (${os.platform()} ${os.release()}-${os.arch()}; Node.js ${process.version})`
 const MAX_BUFFER_LENGTH = buffer.constants.MAX_LENGTH
 const MAX_STRING_LENGTH = buffer.constants.MAX_STRING_LENGTH
+const HEAP_SIZE_LIMIT = require('v8').getHeapStatistics().heap_size_limit
```

Comment:
1. This is limit for Node instead runtime process. This value is static at time of instantiation and configurable through: `NODE_OPTIONS="$NODE_OPTIONS --max-old-space-size=4096"` (MB)
2. If the total memory for OSD (by process) needed exceed this limit, lets say 4GB above, then it will crash as OOM 
`FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory`